bootstrap-joining-demo/aws-api-ecs/README.md

```diff
@@ -0,0 +1,190 @@
+# Your Akka Cluster on ECS
+
+**Note:**
+
+The ECS Service Discovery module and this demo are both written solely
+for use with awsvpc mode networking. Use with bridge mode or host mode
+networking is _not_ supported.
+
+**Note:**
+
+The provided CloudFormation application stack configures the task definition
+and service for use with Fargate. This is done in the interest of simplicity
+(it removes the need for the stack to also create a container instance).
+At the time of writing, Fargate is only available in the us-east-1 region.
+This is why the CloudFormation stacks and scripts have the region set to
+us-east-1.
+
+However, the ECS Service Discovery is _not_ limited to use with Fargate. By
+removing the two Fargate configuration lines and adding a container-instance
+resource you can also run it with "classic" bring-your-own-container-instance
+ECS (and thus in any region).
+
+
+# Step 1: Examine the source code
+
+Examine the source code. See how simple it is. There is no hard coding
+of seed node IPs or host names in the application config.
+
+
+# Step 2: Create the ECR repository
+
+You can do this manually via the web console by creating an ECR repo called
+`bootstrap-joining-demo-aws-api-ecs`, _or_ by running the included
+CloudFormation wrapper script:
+
+`./scripts/create-infrastucture.sh`
+
+
+# Step 3: Build and publish the Docker image
+
+The easiest way to do this is by running the included script (which you read
+earlier, right?):
+
+`./scripts/publish-application.sh`
+
+The script uses `docker:publishLocal` and then tags and pushes the image
+manually (in favour of just using `docker:publish` as might normally be done)
+because this avoids the need to set `dockerRepository` within the SBT build
+(which would then require that we inject the AWS account ID into the build).
+
+
+# Step 4: Identify your subnet IDs
+
+The application stack configures the demo ECS service to launch with five task.
+instances.
+
+Before you can create the cluster itself, you need to identify which subnets
+you want the task instances to be launched in.
+
+Go to the VPC service area of the AWS Management Console, then go to the
+Subnets tab. You can compare with the screenshot below as a guide to what
+you're looking for.
+
+![Identifying subnet IDs](screenshots/identify-subnet-ids.png)
+
+For the purposes of the demo we'll assume you're using your default VPC (the
```

Comment:
:+1: Done. I used to write more formally (and in many cases still do), but in the last couple of years, inspired by [Write Like You Talk](http://www.paulgraham.com/talk.html), I've started deliberately being less formal. Perhaps this is not an appropriate context though, and either way, consistency with the existing conventions trumps individual preferences here. It's only because I'm less tuned in to the conventions that I inadvertently didn't conform :-).